artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/AMQPLargeMessageReader.java

```diff
@@ -50,14 +50,15 @@ public DeliveryAnnotations getDeliveryAnnotations() {
    @Override
    public void close() {
       if (!closed) {
-         if (currentMessage != null) {
-            try {
-               currentMessage.deleteFile();
-            } catch (Throwable error) {
-               ActiveMQServerLogger.LOGGER.errorDeletingLargeMessageFile(error);
-            } finally {
-               currentMessage = null;
+         try {
+            AMQPLargeMessage localCurrentMessage = currentMessage;
```

Comment:
There are various other usages of currentMessage that could probably benefit from similar update.

Reply:
There was actually only addBytes... as the initialization should stay as is.

I amended the PR with the addBytes change.

Reply:
@gemmellr actually... I am doing another change instead.


The AMQPLargeMessage should be created within the session executor.

Right now that's only working because I'm not actually creating a file there yet.. but this is a bit fragile.. as anyone could break it..


I will move the initialization of the new Object within the addBytes..

Reply:
> There was actually only addBytes... as the initialization should stay as is.
> 
> I amended the PR with the addBytes change.

readBytes also reads (and writes) the variable multiple times.

Reply:
not any more after my push -f